File: src/main/java/org/bipmed/brave/cli/BraveImport.java
package org.bipmed.brave.cli;

import org.bipmed.brave.vcf.VcfImporter;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.CommandLineRunner;
import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Component;

@Component
@Profile("!test")
public class BraveImport implements CommandLineRunner {

    @Value("${filename}")
    private String filename;

    @Value("${datasetId}")
    private String datasetId;

    @Value("${assemblyId}")
    private String assemblyId;
    private VcfImporter vcfImporter;

    @Autowired
    public BraveImport(VcfImporter vcfImporter) {
        this.vcfImporter = vcfImporter;
    }

    @Override
    public void run(String... args) {
        if (!filename.isEmpty() && !datasetId.isEmpty() && !assemblyId.isEmpty()) {
            long count = vcfImporter.importVariants(filename, datasetId, assemblyId);
            System.out.println("Imported " + count + " variants.");
        } else {
            System.out.println("USAGE: java -jar brave-import.jar --url http://localhost:8080 --filename file.vcf.gz --datasetId bipmed --datasetId hg19");
            System.exit(1);
        }
    }
}

File: src/main/kotlin/org/bipmed/brave/import/cli/BraveImport.kt
package org.bipmed.brave.import.cli

import org.bipmed.brave.import.vcf.VcfImporter
import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.CommandLineRunner
import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Component
import kotlin.system.exitProcess

@Component
@Profile("!test")
class BraveImport(private val vcfImporter: VcfImporter) : CommandLineRunner {

    @Value("\${filename}")
    val filename: String = ""

    @Value("\${datasetId}")
    val datasetId: String = ""

    @Value("\${assemblyId}")
    val assemblyId: String = ""

    override fun run(vararg args: String) {
        if (filename.isNotBlank() && datasetId.isNotBlank() && assemblyId.isNotBlank()) {
            val count = vcfImporter.import(filename, datasetId, assemblyId)
            println("Imported $count variants.")
        } else {
            println("USAGE: java -jar brave-import.jar --url http://localhost:8080 --filename file.vcf.gz --datasetId bipmed --datasetId hg19")
            exitProcess(1)
        }
    }
}